pkgs/applications/editors/emacs/generic.nix

```
@@ -134,7 +135,9 @@ let emacs = stdenv.mkDerivation (lib.optionalAttrs nativeComp {
     ++ lib.optionals withImageMagick [ imagemagick ]
     ++ lib.optionals (stdenv.isLinux && withX) [ m17n_lib libotf ]
     ++ lib.optional (withX && withGTK2) gtk2-x11
-    ++ lib.optionals (withX && withGTK3) [ gtk3-x11 gsettings-desktop-schemas ]
+    ++ lib.optional (withX && withGTK3) gtk3-x11
+    ++ lib.optional withGTK3 gsettings-desktop-schemas
```

I suppose I should change this in a follow-up cherry to something like

```suggestion
    ++ lib.optional (!withNS && withGTK3) gsettings-desktop-schemas
```

I don't have a good understanding of macOS so I could be wrong. Does this seem reasonable? If so, I'll make an additional PR onto master branch and cherry-pick that onto this branch.